#Statistical Analysis of High-Throughput Genomic and Transcriptomic Data 
Fall/Herbst-semester 2016

## Lectures
Mondays 9.00-9.45, 10.00-10.45 (approx) Y23-G-04

## Exercises
Monday 11.00-11.45 Y11-J-05 

## Lecturers

[Dr. Hubert Rehrauer](http://www.fgcz.ch/the-center/people/rehrauer.html), Group Leader of Genome Informatics at [FGCZ](http://www.fgcz.ch/)  

[Prof. Dr. Mark Robinson](http://www.imls.uzh.ch/research/robinson.html), Assistant Professor of Statistical Genomics, [IMLS](http://www.imls.uzh.ch/index.html), UZH

[Dr. Charlotte Soneson](http://csoneson.github.io/), Postdoctoral Fellow, 
[IMLS](http://www.imls.uzh.ch/index.html), UZH


## Schedule

| Date  | Lecturer | Topic | JC1 | JC2 |
| --- | --- | --- | --- | --- |
| 19.09.2016  | Mark; Hubert  | admin, mol. biology basics, R markdown | | |
| 26.09.2016  | Hubert  | exploratory data analysis | | |
| 03.10.2016  | Mark; Hubert  | interactive technology session  | | |
| 10.10.2016  | Hubert  | NGS intro; mapping  | | |
| 17.10.2016  | Charlotte  | hands-on RNA-seq session  | | |
| 24.10.2016  | Mark  | limma 1  | | |
| 31.10.2016  | Mark  | limma 2  | [Topological Data Analysis Generates High-Resolution, Genome-wide Maps of Human Recombination](http://www.sciencedirect.com/science/article/pii/S2405471216301831) {CS} | x |
| 07.11.2016  | Hubert  | RNA-seq quantification  | [Reliable detection of subclonal single-nucleotide variants in tumour cell populations](http://www.nature.com/articles/ncomms1814) <br> {CB,L-WY} | [A network-based method to evaluate quality of reproducibility of differential expression in cancer genomics studies](http://www.impactjournals.com/oncotarget/index.php?journal=oncotarget&page=article&op=view&path[]=5987&pubmed-linkout=1) <br> {TS, SS} |
| 14.11.2016  | Mark  | edgeR+friends 1  | [Impact of statistical models on the prediction of type 2 diabetes using non-targeted metabolomics profiling](https://www.ncbi.nlm.nih.gov/pubmed/27689004) <br> {FB,SM,CP} | [Capturing Heterogeneity in Gene Expression Studies by Surrogate Variable Analysis] (http://journals.plos.org/plosgenetics/article/asset?id=10.1371/journal.pgen.0030161.PDF) <br> {HW, PR} |
| 21.11.2016  | Mark  | edgeR+friends 2  | [Adjusting batch effects in microarray expression data using empirical Bayes methods](http://biostatistics.oxfordjournals.org/content/8/1/118.long) <br> {KH, SS}  |[A statistical approach for identifying differential distributions in single-cell RNA-seq experiments](https://www.ncbi.nlm.nih.gov/pubmed/27782827) <br> {VS, FH} |
| 28.11.2016  | Hubert  | classification  | [A Method for Checking Genomic Integrity in Cultured Cell Lines from SNP Genotyping Data](http://dx.doi.org/10.1371/journal.pone.0155014) <br> {EP, PCC}  |[Shrinkage estimation of dispersion in Negative Binomial models for RNA-seq experiments with small sample size](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3654711/)<br> {AD, KL, XL} |
| 5.12.2016  | Mark  | epigenomics, DNA methylation  | [Empirical Bayes Analysis of a Microarray Experiment](http://www.tandfonline.com/doi/abs/10.1198/016214501753382129) <br> {GA, IA} | [ Redefining CpG islands using hidden Markov models](http://biostatistics.oxfordjournals.org/content/11/3/499) <br> {FL, JK} |
| 12.12.2016  | Mark  | gene set analysis  | Circular binary segmentation for the analysis of array-based DNA copy number data. <br> Olshen AB, Venkatraman ES, Lucito R, Wigler M. <br> http://www.ncbi.nlm.nih.gov/pubmed/15475419 <br> {KR, MP} |  Data-driven hypothesis weighting increases detection power in genome-scale multiple testing. <br>  Nikolaos Ignatiadis,	Bernd Klaus,	Judith B Zaugg	& Wolfgang Huber. <br> http://www.nature.com/nmeth/journal/v13/n7/full/nmeth.3885.html<br> {SRH,  NH, and LPF}|
| 19.12.2016  | Mark  | single-cell  | [The statistical properties of gene-set analysis](http://www.nature.com/nrg/journal/v17/n6/full/nrg.2016.29.html) <br> {AS, FE, MB} | x |


## Useful Links
[Simply Statistics blog](http://simplystatistics.org/archive/)  
[Getting Genetics Done blog](http://www.gettinggeneticsdone.com/)  

## Course material

You can check out the entire set of course materials with the following command:
```
git clone https://github.com/sta426hs2016/material.git
```  
